src/bulk/unordered.ts

```diff
@@ -20,12 +19,12 @@ export class UnorderedBulkOperation extends BulkOperationBase {
     super(collection, options, false);
   }
 
-  override handleWriteError(callback: Callback, writeResult: BulkWriteResult): boolean {
+  override handleWriteError(writeResult: BulkWriteResult): void {
     if (this.s.batches.length) {
-      return false;
+      return;
```

Comment:
length is always a truthy value while in the for loop, after we finish the array is cleared so we will go to the next line which reuses the logic in the parent class. This is the difference between an ordered operation that is interrupted by an error and an unordered one that settles everything and you check the result for what succeeded and what did not.